src/Bindings/ManualBindings_BlockArea.cpp

```diff
@@ -154,16 +154,27 @@ static int tolua_cBlockArea_FillRelCuboid(lua_State * a_LuaState)
 	cBlockArea * self = nullptr;
 	if (!L.GetStackValues(1, self))
 	{
-		return L.ApiParamError("Cannot read self.");
+		return L.ApiParamError("Cannot read self");
 	}
 	if (self == nullptr)
 	{
-		return L.ApiParamError("Invalid 'self', must not be nil.");
+		return L.ApiParamError("Invalid 'self', must not be nil");
 	}
 
 	// Check and get the overloaded params:
 	cCuboid bounds;
 	auto nextIdx = readCuboidOverloadParams(L, 2, bounds);
+	bounds.Sort();
+	if (!(self->IsValidRelCoords(bounds.p1) && self->IsValidRelCoords(bounds.p2)))
+	{
+		return L.FApiParamError(
+			"The bounds ({0} - {1}) are out of range ({2} - {3})",
+			bounds.p1,
+			bounds.p2,
+			Vector3i(0, 0, 0),
+			(self->GetSize() - Vector3i{1, 1, 1})
+		);
+	}
```

Comment:
This changes the function behavior, previously out-of-bounds coords were supposed to work by auto-clamping them, now the operation is downright refused. I'd have preferred the old way - clamp, possibly with a warning message.

Reply:
This does indeed change the function behaviour, but the previous behaviour was crashing the whole server and now it just generates a Lua error. I wanted this function to be consistent with all the others in the BlockArea API, which also generate an error when out-of-range operations are attempted. Is there some reason why clamping is preferable for this function?

Reply:
I'm from a "best effort" camp. Like many UI toolkits, if the coords are invalid, but still the operation can be done in a "limited fashion", then do it (and possibly warn about it) rather than refuse altogether.
The other functions don't make sense to even try. But with filling a rect, one can argue that clamping is the ideal "limited fashion".

Reply:
I agree best effort is nice but if as bearbin says everything else is strict then consistency is better. Having to piecemeal support cases where plugins pass bad input and expect it to silently work because we never warned them runs the risk of turning into a maintenance headache (see Windows).

Reply:
Okay, overruled.